### PR TITLE
sway: update to 1.0.

### PR DESCRIPTION
**Pull request by Derriick**
Sway 1.0

### CODE REVIEW

**Comment by Derriick** [comment]
No problem, that's why I added [DO NOT MERGE]. I just wanted to propose a template for users who would like to compile and install it themselves, like the old versions like [](https://github.com/void-linux/void-packages/pull/5250). However, there is still a problem for those who have musl.

**Comment by maxice8** [comment]
New release available

**Comment by soullivaneuh** [comment]
> Upstream explicitly asks us to not package this release

Do you please have an explanation or a link to it?

**Comment by maxice8** [comment]
On Sat, Mar 02, 2019 at 09:13:10AM -0800, Sullivan SENECHAL wrote:
> > Upstream explicitly asks us to not package this release
> 
> Do you please have an explanation or a link to it?
> 
> -- 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/8171#issuecomment-468939735

It is on the release notes of every release.

https://github.com/swaywm/sway/releases

**Comment by soullivaneuh** [comment]
I did see just after the comment sorry. I was thinking about an issue here. :wink:

**Comment by jcgruenhage** [comment]
New release candidate has been released

**Comment by jcgruenhage** [comment]
1.0 has been released! It's merge-able now

**Comment by maxice8** [comment]
@jcgruenhage still points to rc version, if you really need 1.0 make a Pull request for it and tag me.

**Comment by Derriick** [comment]
I tried to package version 1.0 but meson seems to be looking for scdoc program in the wrong place (the right one is `/usr/bin/scdoc`):

```
Dependency wlroots found: YES 0.5.0
Fetching value of define "WLR_HAS_XWAYLAND" with dependency wlroots: 1
Dependency scdoc found: YES 1.9.4
Program /usr/local/bin/scdoc found: NO

meson.build:100:1: ERROR:  Program(s) ['/usr/local/bin/scdoc'] not found or not executable
```

I don't know yet how to get meson to search for scdoc in the right place, I will propose a patch as soon as I find out how.

**Comment by Johnnynator** [comment]
@Derriick That issue was in the scdoc package, the *.pc file was wrong. I pushed an fix https://github.com/void-linux/void-packages/commit/c4b9586277a2c2daa9e5f14afa55899526cad970 for it, should be in the repo the next few minutes

**Comment by jcgruenhage** [comment]
@maxice8 I meant from an upstream PoV, not the current state of this PR

File: srcpkgs/sway/template [hunk 1]
```diff
@@ -1,25 +1,22 @@
...
+checksum=24dafd0f1e630e97a5dd47233841adf856b665e2321d6207acfe6b3002d1bc56
 
-pre_configure() {
-	sed -i CMakeLists.txt -e "s;-Werror;-Wno-error;"
-}
 post_install() {
 	vlicense LICENSE
 	# must be setuid without logind!
-	chmod u+s ${DESTDIR}/usr/bin/sway
+	#chmod u+s ${DESTDIR}/usr/bin/sway
```
Comment (dukeinblue) [inline srcpkgs/sway/template]: Should this be uncommented? It is not that every system should have elogind installed in order to use Sway...

Comment (CameronNemo) [reply srcpkgs/sway/template]: I put this there as a place holder. Should be changed to an INSTALL script with setcap, if at all possible.

https://github.com/swaywm/sway/wiki/Running-Sway-without-systemd#direct-via-capabilities

Comment (Derriick) [reply srcpkgs/sway/template]: I place the setcap command in an INSTALL script, but I don't know how to check that the script is running correctly.

**Comment by iterus0** [comment]
[Wiki](https://github.com/swaywm/sway/wiki/Running-Sway-without-systemd#direct-via-capabilities) actually recommends to use setuid instead of capabilities:

> Note: Sway won't drop capabilities, use setuid instead.

**Pull request merged**